client/my-sites/domains/index.js

```diff
@@ -247,9 +247,7 @@ export default function() {
 			domainsController.redirectIfNoSite( '/domains/add' ),
 			domainsController.redirectToAddMappingIfVipSite(),
 			jetPackWarning,
-			domainsController.domainSearch,
-			makeLayout,
-			clientRender
+			domainsController.redirectToDomainSearchSuggestion
```

Comment:
If you wanted to clean the code up a little, uou could just redirect here directly without all the previous steps (site selection, redirections...). The `/domain/add/:domain` route will take care of that.

Also, I suspect that the `jetPackWarning` ([code](https://github.com/Automattic/wp-calypso/blob/master/client/my-sites/controller.js#L450-L456)) handler might be buggy -- it sets `context.primary`, but then doesn't call `next` and `makeLayout` never gets called. Doesn't this mean that a blank page will be rendered?

Or maybe I misunderstand the single tree rendering code -- @simison will know better.

Reply:
@jsnajdr Yep, looks like that needs a fix. I'll PR.

Thanks for the ping!

Reply:
PRed: https://github.com/Automattic/wp-calypso/pull/22785